persist-cli/src/main/java/io/ballerina/persist/cmd/Migrate.java

```diff
@@ -0,0 +1,795 @@
+            } catch (IOException e) {
+                errStream.println("Error: Copying file failed: " + e.getMessage());
+                return;
+            }
+        } else {
+            // Delete the newMigrateDirectory
+            boolean deleteNewMigrateFolder = newMigrateDirectory.delete();
+            if (!deleteNewMigrateFolder) {
+                errStream.println("Error: Failed to delete timestamp folder.");
+            }
+
+            // Delete the migrations directory if it is empty
+            if (migrationsDir.exists() && migrationsDir.isDirectory()) {
+                File[] files = migrationsDir.listFiles();
+                if (files != null && files.length == 0) {
+                    try {
+                        Files.delete(migrationsDir.toPath());
+                    } catch (IOException e) {
+                        errStream.println("Error: Failed to delete migration folder: " + e.getMessage());
+                        return;
+                    }
+                }
+            }
+        }
+    }
+
+    // Create a timestamp folder name
+    private static String createTimestampFolder(String migrationName, File migrationsDir) {
+        Instant currentTime = Instant.now();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd-HH-mm-ss");
+        String timestamp = formatter.format(ZonedDateTime.ofInstant(currentTime, ZoneOffset.UTC));
+
+        String newMigration = migrationsDir + File.separator + timestamp + "_" + migrationName;
+
+        return newMigration;
+    }
+
+    // Create the timestamp directory
+    private static File getTimestampDirectory(String newMigration) {
+        File newMigrateDirectory = new File(newMigration);
+
+        if (!newMigrateDirectory.exists()) {
+            boolean isDirectoryCreated = newMigrateDirectory.mkdirs();
+            if (!isDirectoryCreated) {
+                errStream.println("Error: Failed to create directory: " + newMigrateDirectory.getAbsolutePath());
+            } else {
+                return newMigrateDirectory;
+            }
+        }
+
+        return null;
+    }
+
+    // Create the timestamp directory without returning the directory
+    private static void createTimestampDirectory(String newMigration) {
+        File newMigrateDirectory = new File(newMigration);
+
+        if (!newMigrateDirectory.exists()) {
+            boolean isDirectoryCreated = newMigrateDirectory.mkdirs();
+            if (!isDirectoryCreated) {
+                errStream.println("Error: Failed to create directory: " + newMigrateDirectory.getAbsolutePath());
+            }
+        }
+    }
+
+    private static List<String> findDifferences(Module previousModel, Module currentModel) {
+        List<String> queries = new ArrayList<>();
+
+        List<String> addedEntities = new ArrayList<>();
+        List<String> removedEntities = new ArrayList<>();
+        HashMap<String, List<FieldMetadata>> addedFields = new HashMap<>();
```

Comment:
Can't we use `EntityField` instead of `FieldMetadata`?

Reply:
I created fieldmetadata to be more versatile to due the entityfields causing repeated issues and bugs when I used them before